cockroachdb/templates/_helpers.tpl

```diff
@@ -274,13 +274,18 @@ Validate that if user enabled tls, then either self-signed certificates or certi
 {{- end -}}
 
 {{- define "cockroachdb.securityContext.versionValidation" }}
+{{/* Allow using `securityContext` for custom images. */}}
+{{- if ne "cockroachdb/cockroach" .Values.image.repository -}}
+    {{ print true }}
+{{- else -}}
```

Comment:
The only issue with the approach is that any custom image built with version less than `21.2.13` will have the securityContext enabled and the cockroachdb pods will not come up.

Instead of returning true, we return false from here in custom images case. Also, we can have a flag `force` in `securityContext` section, if enabled  we allow the securityContext insertion in all cases, so custom image user's can enable the `force` flag to insert the security context. This `force` flag is only needed in statefulset security context.

@rail WDYT?

Reply:
Sounds sane to me. I'd rather have an explicit directive to use `securityContext` rather deriving it from the repo name.
Thank you for your contribution @tyranron!

Reply:
@prafull01 on my opinion, the described way is not better, because:
- Custom image of version less than `21.2.13` is more rare situation than a custom images built on top of latest CockroachDB version. Furthermore, as the time goes by, they will become even more and more rare.
- So, in my opinion, it's better to make a minority of users to opt-out `securityContext` explicitly, rather than making a majority of users to opt-in it explicitly.
- Also, because it's easy for the majority of users to skip using `securityContext` without wanting so. The thing is too subtle. You have your deployment being configured, then you just change the image repository and voilà - the `securityContext` is subtly disabled, no errors are printed.
- While the opposite is at least noticeable. If a user uses a custom image built on top of prior `21.2.13` version, the pods won't come up, he will see the error, check the Helm chart and disable the `securityContext`.